### PR TITLE
[RLlib] Various updates to the Release CI RLlib

### PR DESCRIPTION
**Pull request by avnishn**
- change most of the release test instances to use the faster g5 instances. In theory this should allow us to reduce the timeout window for these tests, since they run faster
- disable rl_module/learner_api for ppo with attention or LSTM

Signed-off-by: Avnish <avnishnarayan@gmail.com>

## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [ ] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [x] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Comment by avnishn** [comment]
https://buildkite.com/ray-project/release-tests-pr/builds/43745

**Comment by can-anyscale** [comment]
CI doesn't build ray wheel if you change only files under release. Maybe change something in rllib and undo it later to trigger the ray wheel.

File: release/ray_release/cluster_manager/cluster_manager.py [hunk 1]
```diff
@@ -73,7 +73,7 @@ def set_cluster_env(self, cluster_env: Dict[str, Any]):
             self.cluster_env_name = (
                 f"{self.project_name}_{self.project_id[4:8]}"
                 f"__env__{self.test.get_name().replace('.', '_')}__"
-                f"{dict_hash(self.cluster_env)}"
+                f"{dict_hash(self.cluster_env)}__2164728348"
```
[the remark this answers is not in the document]
Comment (avnishn) [reply release/ray_release/cluster_manager/cluster_manager.py:76]: @can-anyscale suggested I do this for getting cluster envs to rebuild properly.

[the remark this answers is not in the document]
Comment (aslonnie) [reply release/ray_release/cluster_manager/cluster_manager.py:76]: I feel that you did not answer my question :)

[the remark this answers is not in the document]
Comment (can-anyscale) [reply release/ray_release/cluster_manager/cluster_manager.py:76]: That's the number in our bank account in 2 years, let's move fast!

**Comment by can-anyscale** [comment]
@aslonnie just told me that we have stopped building anyscale/ray-ml:nightly-py37 since 15 days ago so it's critical that all ml release tests need to migrate to py38 at this point, cc: @krfricke

Since the branch cut is tomorrow, let's finish this today or tomorrow so we have proper release test signals during branch cut

**Review by kouroshHakha — APPROVED** [review]

**Pull request merged**